addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -682,4 +682,39 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
         }).then(reload);
     },
 });
+
+options.registry.WebsiteSaleProductAttribute = options.Class.extend({
+    /**
+     * @override
+     */
+     willStart: async function () {
+        const _super = this._super.bind(this);
+        this.attributeID = this.$target.closest('[data-attribute_id]').data('attribute_id');
+        return _super(...arguments);
+    },
+
+    /**
+     * @see this.selectClass for params
+     */
+    setDisplayType: function (previewMode, widgetValue, params) {
+        this._rpc({
+            route: '/shop/config/attribute',
+            params: {
+                attribute_id: this.attributeID,
+                display_type: widgetValue,
+            },
+        }).then(reload);
```

Comment:
I wonder if it wouldn't be better to display a message below the <we-select> to tell the user the modification will be visible only after the reload ?

I think the drawback of losing other changes when you're encouraged to reload the page is maybe too strict and not enough intuitive.

Reply:
Side note: As it was validated functionally and it works like that for numerous other options, I think there is no point here.

We can go ahead.

Reply:
It follows the behaviour of some other changes we can do with the web editor, I think's it's ok as is.